src/main/java/address/MainApp.java

```diff
@@ -35,15 +41,29 @@ public void start(Stage primaryStage) {
     }
 
     protected void setupComponents() {
+
         config = getConfig();
         PreferencesManager.setAppTitle(config.appTitle);
-        AddressBookWrapper dataFromFile = StorageManager.getDataFromFile(
-                                                PreferencesManager.getInstance().getPersonFile());
+
+        final File initialFile = PreferencesManager.getInstance().getPersonFile();
+        AddressBookWrapper dataFromFile;
+        Exception fileLoadException = null;
+        try {
+             dataFromFile = StorageManager.loadDataFromSaveFile(initialFile);
+        } catch (JAXBException | FileContainsDuplicatesException e) {
+            fileLoadException = e;
+            dataFromFile = new AddressBookWrapper();
+        }
+
```

Comment:
Right now if the **initial** load detects duplicate data, we will inform the user with an error dialogue (for consistency, the same dialogue as the one triggered when the user tries to **manually** load a file containing duplicates). 
The implementation also uses the same event `FileOpeningExceptionEvent`.

The problem is that the `setupComponents` method has to **wait** till `mainController` is loaded in order to trigger the error dialogue code, and we need to remember the result of the load attempt till `mainController` is ready before posting the appropriate event. I'm not sure how to nicely shift the logic downwards to `StorageManager`, since we have to track both cases 
- [no duplicates => `AddressBookHandler`] 
- [duplicates => event or exception]